### PR TITLE
[Cart][Behat] Properly rounded taxes on order

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets |
| License       | MIT
| Doc PR        |

Added scenario that ensures proper taxes on order and order item units.

### CODE REVIEW

File: features/shop/round_taxes_on_order_item_level.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        And store has "EU VAT" tax rate of 23% for "Clothes" within "EU" zone
+        And store has a product "PHP T-Shirt" priced at "€10.20"
+        And product "PHP T-Shirt" belongs to "Clothes" tax category
+        And store ships everything for free
+        And store allows paying offline
+        And I am logged in as "john@example.com"
+
+    Scenario: Properly rounded taxes for multiple products
+        When I add 1000 products "PHP T-Shirt" to the cart
+        Then my cart total should be "€12,546.00"
```
Comment (pjedrzejewski) [inline features/shop/round_taxes_on_order_item_level.feature]: For customer the precision should be = 2...

File: features/shop/round_taxes_on_order_item_level.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        And there is user "john@example.com" identified by "password123", with "United Kingdom" as shipping country
+        And store has "EU VAT" tax rate of 23% for "Clothes" within "EU" zone
+        And store has a product "PHP T-Shirt" priced at "€10.20"
+        And product "PHP T-Shirt" belongs to "Clothes" tax category
+        And store ships everything for free
+        And store allows paying offline
+        And I am logged in as "john@example.com"
+
+    Scenario: Properly rounded taxes for multiple products
+        When I add 1000 products "PHP T-Shirt" to the cart
```
Comment (michalmarcinkowski) [inline features/shop/round_taxes_on_order_item_level.feature]: I would go with example from my comment [here](https://github.com/Sylius/Sylius/pull/3868#issuecomment-172482735) with quantity 10.

Comment (Zales0123) [inline features/shop/round_taxes_on_order_item_level.feature]: :+1:

File: features/shop/round_taxes_on_order_item_level.feature [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        And store has "EU VAT" tax rate of 23% for "Clothes" within "EU" zone
+        And store has "Low VAT" tax rate of 10% for "Mugs" within "EU" zone
+        And store has a product "PHP T-Shirt" priced at "€10.10"
+        And store has a product "Symfony Mug" priced at "€45.95"
+        And store has a product "PHP Mug" priced at "€45.94"
+        And product "PHP T-Shirt" belongs to "Clothes" tax category
+        And product "Symfony Mug" belongs to "Mugs" tax category
+        And product "PHP Mug" belongs to "Mugs" tax category
+        And store ships everything for free
+        And store allows paying offline
```
Comment (michalmarcinkowski) [inline features/shop/round_taxes_on_order_item_level.feature]: I suppose we don't need the two steps above

File: features/shop/round_taxes_on_order_item_level.feature [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    In order to avoid taxes amount inaccuracy
+    As a Customer
+    I want to have correct taxes applied to my order
+
+    Background:
+        Given the store is operating on a single "France" channel
+        And there is "EU" zone containing all members of European Union
+        And default currency is "EUR"
+        And default tax zone is "EU"
+        And there is user "john@example.com" identified by "password123", with "United Kingdom" as shipping country
```
Comment (michalmarcinkowski) [inline features/shop/round_taxes_on_order_item_level.feature]: `with "United Kingdom" as shipping country` isn't it redundant information?

Comment (pjedrzejewski) [inline features/shop/round_taxes_on_order_item_level.feature]: Why redundant? I'd just move it to another step and say:

```
And their shipping address is to "United Kingdom"
```

Comment (michalmarcinkowski) [inline features/shop/round_taxes_on_order_item_level.feature]: Redundant because it's not important for these scenarios

**Comment by michalmarcinkowski** [comment]
:+1:

**Pull request merged**